config/routes.rb

```diff
@@ -72,5 +72,17 @@
   get "/world/*taxon_base_path", to: "world_wide_taxons#show"
   get "/brexit(.:locale)", to: "transition_landing_page#show"
   get "/transition(.:locale)", to: "transition_landing_page#show"
+
+  # We get requests for URLs like
+  # https://www.gov.uk/topic%2Flegal-aid-for-providers%2Fmake-application%2Flatest
+  # which fall through to here and error in the taxons controller.
+  # We can fix the path and redirect to the correct place.
+  get "/:slug",
+    to: redirect { |path_params, req|
+      query_string = req.query_string.present? ? "?" + req.query_string : ""
+      req.path.gsub("%2F", "/") + query_string
```

Comment:
You could do `[req.path.gsub("%2F", "/"), req.query_string].flatten.join("?")` here, or perhaps use `CGI.unescape req.fullpath` but that'd also convert the query params

Reply:
That's much nicer, thanks!

Reply:
Turns out that an empty `query_string` is `""`, not `nil`, so the compact thing doesn't work quite right.